sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1498,6 +1498,45 @@ def test_neighbors_validate_parameters(Estimator):
         nbrs.predict([[]])
 
 
+@pytest.mark.parametrize(
+    "Estimator",
+    [
+        neighbors.KNeighborsClassifier,
+        neighbors.RadiusNeighborsClassifier,
+        neighbors.KNeighborsRegressor,
+        neighbors.RadiusNeighborsRegressor,
+    ],
+)
+def test_neighbors_non_metric_minkowski(Estimator):
+    """
+    Validation of all classes extending NeighborsBase for minkowski metric with p < 1
+    """
+    X = rng.random_sample((10, 2))
+    y = np.ones(10)
+
+    model = Estimator(p=0.1)
```

Comment:
I would make it explicit that `algo="auto"` is used here.

```suggestion
    model = Estimator(p=0.1, algo="auto")
```

and add another case for:

```python
    model = Estimator(p=0.1, algo="brute")
```